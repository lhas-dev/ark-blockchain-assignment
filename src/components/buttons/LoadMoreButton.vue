<template>
  <button
    @click="onClick"
    class="mx-auto block my-6 p-3 text-gray-500 hover:text-gray-600 border rounded border-gray-500 hover:border-gray-600"
  >Load More</button>
</template>

<script lang="ts">
import Vue from "vue";

const LoadMoreProps = Vue.extend({
  props: {
    onClick: {
      type: Function,
      required: true
    }
  }
});

export default class FlatButton extends LoadMoreProps {}
</script>

<style></style>
