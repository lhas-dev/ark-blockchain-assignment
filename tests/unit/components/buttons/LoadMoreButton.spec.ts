import { shallowMount } from "@vue/test-utils";
import LoadMoreButton from "@/components/buttons/LoadMoreButton.vue";

describe("LoadMoreButton.vue", () => {
  it("renders correctly", () => {
    const wrapper = shallowMount(LoadMoreButton, {
      stubs: ["router-link", "router-view"]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
