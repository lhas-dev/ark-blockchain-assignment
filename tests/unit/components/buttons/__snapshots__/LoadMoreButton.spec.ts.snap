// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadMoreButton.vue renders correctly 1`] = `
<button class="mx-auto block my-6 p-3 text-gray-500 hover:text-gray-600 border rounded border-gray-500 hover:border-gray-600">
  Load More
</button>
`;
